### PR TITLE
Persistent DHT

### PR DESCRIPTION
**Pull request by tok-kkk**
- User persistent storage(database) for the `MultiAddrStore`
- Improve the p2p bootstrap logic

### CODE REVIEW

**Comment by tok-kkk** [comment]
[Persist peer addresses in Lightnode](https://trello.com/c/CYdHXBTT/142-persist-peer-addresses-in-lightnode)

File: lightnode.go [hunk 1]
```diff
@@ -156,8 +156,10 @@ func New(ctx context.Context, options Options, logger logrus.FieldLogger, sqlDB
 	}
 
 	// Initialise the multi-address store.
-	table := kv.NewTable(kv.NewMemDB(kv.JSONCodec), "addresses")
-	multiStore := store.New(table, options.BootstrapAddrs)
+	multiStore, err := store.New(sqlDB, options.BootstrapAddrs)
+	if err != nil {
+		logger.Panicf("fail to initialize multiAddress store, err = %v", err)
```
Comment (jazg) [inline lightnode.go:161]:
```suggestion
		logger.Panicf("failed to initialise multi-address store: %v", err)
```

File: store/store.go [hunk 2]
```diff
@@ -1,108 +1,249 @@
...
 
-// MultiAddrStore is a store of `addr.MultiAddress`es.
-type MultiAddrStore struct {
-	store          db.Table
+const TableNameAddresses = "addresses"
+
+// MultiAddrStore stores darknode MultiAddresses.
+type MultiAddrStore interface {
+
+	// Upsert the MultiAddress if exists, otherwise insert it into the store.
```
Comment (jazg) [inline store/store.go:18]:
```suggestion
	// Insert the MultiAddress if it does not exist, otherwise update it.
```

**Review by jazg — APPROVED** [review]
Looks good!

**Pull request merged**